command/state.go

```diff
@@ -274,3 +282,30 @@ func remoteStateFromPath(path string, refresh bool) (*state.CacheState, error) {
 
 	return remoteState(localState, path, refresh)
 }
+
+// This is a scary but not-particularly-helpful error message. This situation
+// should not arise in legitimate usage, but if it *does* arise there isn't
+// really a "magic bullet" to fix it, so this message is primarily to explain
+// the gravity of the situation to the user.
+//
+// In future it would be nice to do more to prevent this situation from
+// arising. In medium-term, we could provide more elaborate advice on the
+// website and link to it here. For now, we're just focused on minimizing the
+// risk of accidental loss of state, so we do our best.
+const cacheRefreshConflictMessage = `
+This may mean that you have activated remote state while you already had
+resources locally, but the remote state also has resources. In this case,
+you must decide to keep either the local or remote resources.
+
+This may alternatively mean that two "refresh" or "apply" operations
+ran concurrently and have created divergent states. In this case, some
+resources may be duplicated across the two states.
+
+In either case, caution is advised. Your local state cache is preserved
+at .terraform/terraform.tfstate and it will not overwrite the remote until
+the confict is resolved.
+
+At this point it is safe to disable remote state, which will retain the
+local state on this computer and leave the remote state untouched:
+    terraform remote config -disable -pull=false
```

Comment:
I would reword this error message to provide more actions. My questions as a user would be:

* I definitely want my local state to override the remote state, what do I do?

* I definitely want my remote state to override my local state, what do I do?